### PR TITLE
Milestone3: Bookstore use Redux in React components

### PR DESCRIPTION
**Pull request by Chandan-devs-tech**
## In this milestone I have :
- Wrapped `<App>` with the `<Provider>` component from react-redux
- Passed configured store into the `<Provider>`
- Created an initialState variable for `bookSlice`,
- Displayed books, received from the slice, in the reusable components
- Dispatched actions using `useDispatch`
- Added a `<Button>` component, which includes:
  - An event handler that adds a book to the books array (with attributes id, title, and author)
- Added a `<Button>` component, which includes:
  - An event handler that removes a book from the books array (by id)

### CODE REVIEW

File: src/components/Input.jsx [hunk 1]
```diff
@@ -1,14 +1,57 @@
...
-);
+const Input = () => {
+  // <div>
+  // <h2>Add new book</h2>;
+  //   <form>
+  //     <input type="text" name="" id="" placeholder="Book Title" />
+  //     <input type="text" name="" id="" placeholder="Author" />
+  //     <button type="button">Add Book</button>
+  //   </form>
+  // </div>
```
Comment (V-Blaze) [inline src/components/Input.jsx]: - You have done great so far :+1: :+1: Your redux store and project work as expected. But there are some best practice requirements you need to meet before moving to the next milestone. :+1: :100: 
- [ ] Following the best practices, you are not meant to leave unused code as a comment on your code base. I'll suggest you remove this as it makes your codebase look a bit a bit unprofessional. :+1: :smile:

File: src/components/Input.jsx [hunk 2]
```diff
@@ -1,14 +1,57 @@
...
+          value={title}
+          onChange={(e) => setTitle(e.target.value)}
+        />
+        <input
+          type="text"
+          placeholder="Author"
+          value={author}
+          onChange={(e) => setAuthor(e.target.value)}
+        />
+        {/* <button type="submit">Add Book</button> */}
```
Comment (V-Blaze) [inline src/components/Input.jsx]: - [ ] Just as I pointed out above about Following the best practices, you are not meant to leave unused code as a comment on your code base. I'll suggest you remove this as it makes your codebase look a bit a bit unprofessional. :+1: :smile:

File: src/components/Books.jsx [hunk 3]
```diff
@@ -1,25 +1,14 @@
 import React from 'react';
-import { v4 as uuidv4 } from 'uuid';
-import BookItems from './BookItems';
+// import { v4 as uuidv4 } from 'uuid';
+// import BookItems from './BookItems';
```
Comment (V-Blaze) [inline src/components/Books.jsx]: - [ ] Just as I pointed out above about Following the best practices, you are not meant to leave unused code as a comment on your code base. I'll suggest you remove this as it makes your codebase look a bit a bit unprofessional. :+1: :smile:

File: src/components/BookList.jsx [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+// import AddButton from './AddButton';
+
+function BookList() {
+  const booksData = useSelector((state) => state.book.books);
+  return (
+    <div>
+      {booksData.map((book) => (
+        <BookItems key={book.id} book={book} />
+      ))}
+      {/* <AddButton /> */}
```
Comment (V-Blaze) [inline src/components/BookList.jsx]: - [ ] Just as I pointed out above about Following the best practices, you are not meant to leave unused code as a comment on your code base. I'll suggest you remove this as it makes your codebase look a bit a bit unprofessional. :+1: :smile:

File: src/components/BookList.jsx [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import { useSelector } from 'react-redux';
+import BookItems from './BookItems';
+// import AddButton from './AddButton';
```
Comment (V-Blaze) [inline src/components/BookList.jsx]: - [ ] Just as I pointed out above about Following the best practices, you are not meant to leave unused code as a comment on your code base. I'll suggest you remove this as it makes your codebase look a bit a bit unprofessional. :+1: :smile:

File: src/App.js [hunk 6]
```diff
@@ -12,6 +12,7 @@ function App() {
         <Route path="/" element={<Books />} />
         <Route path="/categories" element={<Categories />} />
       </Routes>
+      {/* <Books /> */}
```
Comment (V-Blaze) [inline src/App.js]: - [ ] Just as I pointed out above about Following the best practices, you are not meant to leave unused code as a comment on your code base. I'll suggest you remove this as it makes your codebase look a bit a bit unprofessional. :+1: :smile:

**Review by V-Blaze — CHANGES_REQUESTED** [review]
## Hi @Chandan-devs-tech  , 

## Required changes 🖊️ 
# Almost there 🟡 🟦 🟨 

giphy


Great work so far, You have come a long way, keep moving forward👏 👍  
You've done well implementing the project requirements, but there are still some issues I think you need to improve on. 😄 👋 


 ## Highlights
- Redux store setup correctly✔
-  Adding and removing books works as expected👋 
-  Correct Git flow✔

 ## Required Changes ♻️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by victorokoroji — APPROVED** [review]
Hi @Chandan-devs-tech , 

# ✔️   Status: APPROVED 🏆 🎉 🎊 

You have done a great job on this project, keep it up 👍 🥇. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
congratulations!  🎉💯🌟 

## Top highlight 🎯

- Requested change implemented ✅ 



## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**